edifice/hooks.py

```diff
@@ -786,16 +787,298 @@ def expensive_computation():
         bignumber = use_memo(expensive_computation, (x_factor,))
 
     Args:
-        fn: A function of no arguments which returns a value.
-        dependencies: The value will be recomputed when the dependencies change. If :code:`None`, then the value will be recomputed every render.
+        fn:
+            A function of no arguments which returns a value.
+        dependencies:
+            The value will be recomputed when the dependencies change.
+            If :code:`None`, then the value will be recomputed every render.
     Returns:
         The memoized value from calling :code:`fn`.
 
     """
-    stored, _ = use_state([None,None])
+    stored, _ = use_state([None, None])
     if dependencies is None:
-        stored[0] = fn() # type: ignore  # noqa: PGH003
+        stored[0] = fn()  # type: ignore  # noqa: PGH003
     elif stored[1] != dependencies:
-        stored[0] = fn() # type: ignore  # noqa: PGH003
+        stored[0] = fn()  # type: ignore  # noqa: PGH003
         stored[1] = dependencies
-    return stored[0] # type: ignore  # noqa: PGH003
+    return stored[0]  # type: ignore  # noqa: PGH003
+
+
+_T_context = tp.TypeVar("_T_context")
+
+_edifice_context = cast(dict[str, tuple[Any, dict[str, Callable]]], {})
```

Comment:
```suggestion
_edifice_context = cast(dict[str, tuple[Any, tuple[Callable,...]]], {})
```

I don't think we need to use the `RenderContext.current_value` as a key for a dictionary of setter functions.

Instead we can just store the setter functions in a `tuple[Callable,...]`. (I suppose we could use a `set[Callable]` if we want to get fancy.)